### ISSUE TITLE
Most of notification data in eventListener returned as undefined

### ISSUE THREAD

**Issue by lukebars** [react-native-push-notification/ios#147]
Hey.

I've currently done full integration as per documentation (AppDelegate .h and .m), and notification event listener returns most of the data as undefined. 

```js
    RNCPushNotificationIOS.addEventListener('localNotification', notif => {
      console.log(notif);
    });
// {"_alert": undefined, "_badgeCount": undefined, "_category": "", "_data": {"id": "1231231"}, "_fireDate": undefined, "_isRemote": undefined, "_remoteNotificationCompleteCallbackCalled": false, "_sound": undefined}
```

Although in the native side you can easily get fireDate etc., it results undefined in the listener.  

```objective-c
UIApplication *app = [UIApplication sharedApplication];
NSArray *eventArray = [app scheduledLocalNotifications];
for (int i=0; i<[eventArray count]; i++)
{
    UILocalNotification* oneEvent = [eventArray objectAtIndex:i];
    NSLog(@"notifs hello %@", oneEvent);
}
// notifs hello <UIConcreteLocalNotification: 0x600003561e00>{fire date = Wednesday, July 8, 2020 at 12:37:21 PM Eastern European Summer Time, time zone = (null), repeat interval = 0, next fire date = Wednesday, July 8, 2020 at 12:37:21 PM Eastern European Summer Time, user info = {
```
Notifications work alright and fire on time with message and text. What am I missing?

```js
    RNCPushNotificationIOS.scheduleLocalNotification({
      fireDate: new Date(Date.now() + 15000).toISOString(),
      alertTitle: 'hello',
      alertBody: 'message 2',
      userInfo: { id: 1231231 },
      alertAction: 'view',
    });
```

**Comment by lukebars** [comment]
This is happening because from native side we do get

```
 data {"body": "message 2", "category": "", "date": "2020-07-08T14:01:34.374+03:00", "identifier": "54D00615-D28C-4DDD-940D-D53A84AAEA4C", "thread-id": "", "title": "hello", "userInfo": {"id": 1231231}}
```

and in `@react-native-community/push-notification-ios/js/index.js:398` there are wrong fields being accessed ! :

```
else {
      // Local notifications aren't being sent down with `aps` dict.
      this._badgeCount = nativeNotif.applicationIconBadgeNumber;
      this._sound = nativeNotif.soundName;
      this._alert = nativeNotif.body;
      this._data = nativeNotif.userInfo;
      this._category = nativeNotif.category;
      this._fireDate = nativeNotif.date;
    }
```

@Naturalclar do you have any ideas why this is done this way?

**Comment by lukebars** [comment]
Turns out it was firebase messing arround with notifications

**Comment by 1st8** [comment]
@lukebars Could you kindly provide some hints on what you did to solve this? We are having the exact same issue. Cheers!

**Comment by lukebars** [comment]
@1st8 it was `AppDelegate.m` setup.

Make sure your `AppDelegate.m` matches this: 

```objective-c

- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
{
   ...
  // Define UNUserNotificationCenter
  UNUserNotificationCenter *center = [UNUserNotificationCenter currentNotificationCenter];
  center.delegate = self;
   ...
}

// Required to register for notifications
- (void)application:(UIApplication *)application didRegisterUserNotificationSettings:(UIUserNotificationSettings *)notificationSettings
{
 [RNCPushNotificationIOS didRegisterUserNotificationSettings:notificationSettings];
}
// Required for the register event.
- (void)application:(UIApplication *)application didRegisterForRemoteNotificationsWithDeviceToken:(NSData *)deviceToken
{
 [RNCPushNotificationIOS didRegisterForRemoteNotificationsWithDeviceToken:deviceToken];
}
// Required for the notification event. You must call the completion handler after handling the remote notification.
- (void)application:(UIApplication *)application didReceiveRemoteNotification:(NSDictionary *)userInfo
fetchCompletionHandler:(void (^)(UIBackgroundFetchResult))completionHandler
{
  [RNCPushNotificationIOS didReceiveRemoteNotification:userInfo fetchCompletionHandler:completionHandler];
}
// Required for the registrationError event.
- (void)application:(UIApplication *)application didFailToRegisterForRemoteNotificationsWithError:(NSError *)error
{
 [RNCPushNotificationIOS didFailToRegisterForRemoteNotificationsWithError:error];
}
// Required for the localNotification event.
- (void)application:(UIApplication *)application didReceiveLocalNotification:(UILocalNotification *)notification
{
 [RNCPushNotificationIOS didReceiveLocalNotification:notification];
}

//Called when a notification is delivered to a foreground app.
-(void)userNotificationCenter:(UNUserNotificationCenter *)center willPresentNotification:(UNNotification *)notification withCompletionHandler:(void (^)(UNNotificationPresentationOptions options))completionHandler
{
  completionHandler(UNAuthorizationOptionSound | UNAuthorizationOptionAlert | UNAuthorizationOptionBadge);
}
- (NSURL *)sourceURLForBridge:(RCTBridge *)bridge
{
#if DEBUG
  return [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index" fallbackResource:nil];
#else
  return [[NSBundle mainBundle] URLForResource:@"main" withExtension:@"jsbundle"];
#endif
}

@end
```

**Comment by 1st8** [comment]
@lukebars Thanks for your reply.
I don't think that removing firebase completely is an option though.
Or how do you receive remote notifications now?

**Comment by lukebars** [comment]
> @lukebars Thanks for your reply.
> I don't think that removing firebase completely is an option though.
> Or how do you receive remote notifications now?

Remote notifications are still working for me. I might be wrong that you need to remove firebase stuff, but make sure that your notification related functions match those that I've wrote and test it out.

**Comment by 1st8** [comment]
I was not able to find a fix for this. 

My code is matching yours (and the docs) in the relevant parts, with the addition of https://rnfirebase.io/#configure-firebase-with-ios-credentials
I tried the Firebase `didFinishLaunchingWithOptions` additions both before and the additions for this package.

There is also this issue https://github.com/zo0r/react-native-push-notification/issues/1488 but I think this problem is better discussed here.

Could you reopen this issue?

**Comment by lukebars** [comment]
Hi @1st8 . 

Does your AppDelegate contain anything else firebase related than this?
```
#import <Firebase.h>
...

  if ([FIRApp defaultApp] == nil) {
     [FIRApp configure];
   }
```

**Comment by 1st8** [comment]
Nope, it does not @lukebars 

Thanks for following up, really appreciate it!

**Comment by lukebars** [comment]
Reopened @1st8

**Comment by Dallas62** [comment]
Hi,

I think there is a property name flipping in the library depending on user-interaction:


https://github.com/react-native-community/push-notification-ios/blob/master/ios/RNCPushNotificationIOS.m#L94

```objectivec

  formattedLocalNotification[@"alertAction"] = RCTNullIfNil(notification.alertAction);
  formattedLocalNotification[@"alertTitle"] = RCTNullIfNil(notification.alertTitle);
  formattedLocalNotification[@"alertBody"] = RCTNullIfNil(notification.alertBody);
  formattedLocalNotification[@"applicationIconBadgeNumber"] = @(notification.applicationIconBadgeNumber);
  formattedLocalNotification[@"category"] = RCTNullIfNil(notification.category);
  formattedLocalNotification[@"repeatInterval"] = @(notification.repeatInterval);
  formattedLocalNotification[@"soundName"] = RCTNullIfNil(notification.soundName);
  formattedLocalNotification[@"userInfo"] = RCTNullIfNil(RCTJSONClean(notification.userInfo));
  formattedLocalNotification[@"remote"] = @NO;

```

https://github.com/react-native-community/push-notification-ios/blob/master/ios/RNCPushNotificationIOS.m#L121

```objectivec
  formattedNotification[@"title"] = RCTNullIfNil(content.title);
  formattedNotification[@"body"] = RCTNullIfNil(content.body);
  formattedNotification[@"category"] = RCTNullIfNil(content.categoryIdentifier);
  formattedNotification[@"thread-id"] = RCTNullIfNil(content.threadIdentifier);
  formattedNotification[@"userInfo"] = RCTNullIfNil(RCTJSONClean(content.userInfo));
```

Data returned in the JS part, local-notification pressed by user, is:

```json
{
    "body": "My Notification Message",
    "category": "",
    "date": "2020-08-03T19:43:56.622+02:00",
    "identifier": "D0386218-CEE1-42D2-AD92-62A295631124", 
    "thread-id": "",
    "title": "Local Notification", 
    "userInfo": {"id": "1", "screen": "home"}
}
```

A quick fix can be made in the JS part, but is it the right thing to do ?

https://github.com/react-native-community/push-notification-ios/blob/master/js/index.js#L400

```js
      this._badgeCount = nativeNotif.applicationIconBadgeNumber;
      this._sound = nativeNotif.soundName;
      this._alert = nativeNotif.alertBody || nativeNotif.body;
      this._title = nativeNotif?.alertTitle || nativeNotif?.title;
      this._data = nativeNotif.userInfo;
      this._category = nativeNotif.category || nativeNotif.categoryIdentifier;
      this._fireDate = nativeNotif.fireDate || nativeNotif.date;
```

One more thing, not tested, but I think there is also a mistake on the remote part:

```js
          this._alert = notifVal.alert;
          this._title = notifVal?.alertTitle;
```

When nothing in the objective-c provide a property call `alert` it's `body` or `alertBody`.

Best regards,

**Comment by lukebars** [comment]
Hey, @Naturalclar, can you share your thoughts on this?

**Comment by fedu44** [comment]
It was working 2 weeks ago, but I think I was using a previous version, not the last.

**Comment by nbolender** [comment]
This was fixed in #175 as long as you have updated your AppDelegate to use UNUserNotificationCenter
